### DISCUSSION TITLE
Add ad blocker, auutomatically delete history and manually proxy settings

### DISCUSSION THREAD

**Discussion by NiKola-UE** [Eloston/ungoogled-chromium#2936] [Ideas]
The Brave Browser already has options to automatically delete history after exit and strongly block ads and trackers, but adding manual proxy settings can be challenging as almost no Chromium-based browser has this.

**Comment by networkException** [comment]
Unless not possible by using extensions, ungoogled-chromium wont ship an ad blocker by default.

For history deletion I don't see a reason not to use incognito tabs.

**Comment by NiKola-UE** [comment]
Thank you for fast reaction. 











I don't mind manually adding ad blockers, but I really don't like browser history saving at all, even less if it's constant; although there are special extensions for that too. As I said, Brave already has an option to automatically delete all history immediately after closing, and not even show it in the address bar, so maybe that can be implemented because Brave is FOSS, besides being fast and private. 









Setting up a proxy manually is a bit more difficult, but still not impossible.

**Reply by PF4Public** [reply]
https://github.com/ungoogled-software/ungoogled-chromium/blob/master/docs/flags.md#:~:text=ClearDataOnExit,data%20on%20exit.

**Comment by NiKola-UE** [comment]
OK, can this works only for history, not for cookies and other data?

**Reply by opzch** [reply]
Set the following policy or a registry key under Software\Policies\Chromium

https://admx.help/?Category=Chrome&Policy=Google.Policies.Chrome::ClearBrowsingDataOnExitList
https://chromeenterprise.google/policies/#ClearBrowsingDataOnExitList

![image](https://github.com/user-attachments/assets/2040b0ea-d977-4d38-bb55-b675dae0fb3b)

**Comment by NiKola-UE** [comment]
That's all well and good, but isn't it much easier and simpler if the options for this exist in the browser itself, as is already the case with Brave, Vivaldi or, ewen, Microsoft Edge, which does not require any key or anything like that at all?

**Reply by opzch** [reply]
That's the reason they are made. Although these options already exist in vanilla Chromium and there are several ways to configure them, those who find it too complex end up with one of these browsers (while sacrificing some privacy to the developer company).

Also, if a "key" is added for any of these browsers, it will be given priority and replace/override the option in the browser itself.

**Comment by NiKola-UE** [comment]
To understand. It is likely that for Brave and Vivaldi there were significantly larger teams than behind the one (there is no point in mentioning the Edge) who had enough resources to make significant changes in their browsers. However, I believe there are also reasons why this is not done in this browser. If it is due to a lack of funds, that can certainly change. But if it is because of the philosophy itself, then it is clear that everything will remain as it is, which further means that such changes in this sense should not even be expected and that is perfectly fine.
